core/pouch/migrations.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * @module core/pouch/migrations
+ * @flow
+ */
+
+const { PouchError } = require('.')
+
+/*::
+import type { Pouch } from './'
+import type { Metadata } from '../metadata'
+
+type SchemaVersion = number
+
+type MigrationData = {
+  migrated: string[],
+  errors: PouchError[]
+}
+type MigrationNoop = MigrationData & {
+  type: 'MigrationNoop'
+}
+type MigrationComplete = MigrationData & {
+  type: 'MigrationComplete',
+}
+type MigrationFailed = MigrationData & {
+  type: 'MigrationFailed',
+}
+type MigrationResult = MigrationNoop|MigrationComplete|MigrationFailed
+
+export type Migration = {
+  baseSchemaVersion: SchemaVersion,
+  targetSchemaVersion: SchemaVersion,
+  description: string,
+  affectedDocs: (Metadata[]) => Metadata[],
+  run: (Metadata[]) => Metadata[]
+}
+*/
+
+const SCHEMA_DOC_ID = '_local/schema'
+const SCHEMA_INITIAL_VERSION = 0
+const INITIAL_SCHEMA = {
+  _id: SCHEMA_DOC_ID,
+  version: SCHEMA_INITIAL_VERSION
+}
+const MIGRATION_RESULT_NOOP = 'MigrationNoop'
+const MIGRATION_RESULT_COMPLETE = 'MigrationComplete'
+const MIGRATION_RESULT_FAILED = 'MigrationFailed'
+
+const migrations /*: Migration[] */ = []
+
+function MIGRATION_NOOP() {
```

Comment:
I'm not sure it is very idiomatic to call a function with a name all in upper case